packages/mint-components/src/components/sqm-task-card/sqm-task-card.tsx

```diff
@@ -0,0 +1,216 @@
+ */
+@Component({
+  tag: "sqm-task-card",
+  shadow: true,
+})
+export class TaskCard {
+  @State()
+  ignored = true;
+
+  /**
+   * @uiName Reward Amount
+   */
+  @Prop()
+  rewardAmount: string = "0";
+
+  /**
+   * @uiName Reward Unit
+   */
+  @Prop()
+  rewardUnit: string = "Points";
+
+  /**
+   * @uiName Title Text
+   */
+  @Prop()
+  cardTitle: string = "Title Text";
+
+  /**
+   * @uiName Description Text
+   */
+  @Prop()
+  description: string = "";
+
+  /**
+   * @uiName Goal Repeatable
+   */
+  @Prop()
+  repeatable: boolean = false;
+
+  /**
+   * The number of times a reward can be earned.  Use zero for no limit (the default).
+   *
+   * @uiName Repeat Amount
+   */
+  @Prop()
+  finite: number = 0;
+
+  /**
+   * @uiName Show Progress Bar
+   */
+  @Prop()
+  showProgressBar: boolean = false;
+
+  /**
+   * @uiName Goal Completion Number
+   */
+  @Prop()
+  goal: number = 1;
+
+  /**
+   * @uiName Progress Bar Steps
+   */
+  @Prop()
+  steps: boolean = false;
+
+  /**
+   * @uiName Progress Bar Unit
+   */
+  @Prop()
+  progressBarUnit: string = "";
+
+  /**
+   * @uiName Expiry Date Message
+   */
+  @Prop() expiryMessage: string = "Ends {endDate}";
+
+  /**
+   * @uiName Reward Duration
+   * @uiWidget DateRange
+   * @uiOptions {"allowPastDates":true, "months": 1}
+   */
+  @Prop() rewardDuration: string;
+
+  /**
+   * Shown to users before the start of the task duration.
+   *
+   * @uiName Start Date Message
+   */
+  @Prop() startsOnMessage: string = "Starts {startDate}";
+
+  /**
+   * Timeframe that the task card will be shown
+   *
+   * @uiName Display Duration
+   * @uiWidget DateRange
+   * @uiOptions {"allowPastDates":true, "months": 1}
+   */
+  @Prop() displayDuration: string;
+
+  /**
+   * Shown to users after the end of the task duration.
+   *
+   * @uiName Ended Date Message
+   */
+  @Prop() endedMessage: string = "Ended {endDate}";
+
+  /**
+   * Displays the amount of times that a user has completed a repeatable task.
+   *
+   * @uiName Completed Count Text
+   */
+  @Prop() completedText: string =
+    "Completed {finite, select, 0 {{count, plural, =1 {{count} time} other {{count} times}}} other {{count}/{finite} times}}";
+
+  /**
+   * @uiName CTA Button Text
+   */
+  @Prop()
+  buttonText: string = "Complete Action";
+
+  /**
+   * @uiName CTA Button Link
+   */
+  @Prop()
+  buttonLink: string = "https://example.com/";
+
+  /**
+   * @uiName CTA Button Link Open in New Tab
+   */
+  @Prop()
+  openNewTab: boolean = false;
+
+  /**
+   * Select what type of stat to display for the goal. Manual paths are also supported.
+   *
+   * @uiWidget StatTypeSelectWidget
+   * @uiName Goal Progress Source
+   * @uiOptions {"version": 1.1}
+   */
+  @Prop()
+  statType: string = "/programGoals/count/Referral-Started%2Freferrals";
```

Comment:
@00salmon I am thinking we might actually want to remove this default, thoughts?

Reply:
I'm pretty indifferent, that goal is one that will exist by default for anyone using a referral program i think, but if they are loyalty only it wouldn't be so useful 🤔

Reply:
I'm wondering if we actually want to follow a more fail fast approach

Reply:
we do have a fail-fast component for required props 👀